driver-core/src/main/com/mongodb/internal/connection/DefaultServer.java

```diff
@@ -259,8 +304,28 @@ public void onResult(final T result, final Throwable t) {
         @Override
         public void stateChanged(final ChangeEvent<ServerDescription> event) {
             ServerDescription oldDescription = description;
-            description = event.getNewValue();
-            serverListener.serverDescriptionChanged(new ServerDescriptionChangedEvent(serverId, description, oldDescription));
+            if (shouldReplace(oldDescription, event.getNewValue())) {
+                description = event.getNewValue();
+                serverListener.serverDescriptionChanged(new ServerDescriptionChangedEvent(serverId, description, oldDescription));
+            }
+        }
+
+        private boolean shouldReplace(final ServerDescription oldDescription, final ServerDescription newDescription) {
```

Comment:
Is this the inverse of isStale?

Reply:
Not quite, unfortunately.  It's because the processId comparison is not strictly ordered.  We just compare for inequality.